### DISCUSSION TITLE
[URGENT HELP NEEDED!] Idle anycable-rails occupying too much memory?

### DISCUSSION THREAD

**Discussion by arpansac** [anycable/anycable#130] [General]
My rails 6.0 app with anycable-rails 1.0.0 (front end is in Angular) is taking up 151mb when anycable-rails is started using `bundle-exec anycable` 

With just 50 users we are going beyong 300% CPU usage.
How to debug.
I've put RAILS_ENV=production in the environment

Also, I'm running anycable and go on a different server than puma

**Comment by Envek** [comment]
Memory consumption of anycable RPC doesn't depend on number of connected idle clients. It depends on number of connections/disconnections or number of incoming websocket requests from client. When clients are connected idle or receive messages from server, anycable RPC doesn't do anything.

Please check your app's logic for freshly connected clients or incoming messages handling–most probably you have unoptimized code in your application logic (like huge ActiveRecord result sets).

**Reply by arpansac** [reply]
Thanks for this!
Any specific place where I could read more on optimizing what datasets can anycable/actioncable handle? 
Grateful for the help

**Reply by Envek** [reply]
There's nothing specific for cables here. If you load 1GB of something into memory in your code you will get memory bloat, no matter whether it is insida some rails action in puma, job in sidekiq, or cable channel in anycable-rpc. Try tools like [memory_profiler](https://github.com/SamSaffron/memory_profiler).

**Reply by arpansac** [reply]
Thank you again!
I did a further debug, my code doesn't seem to have any huge active record object lists, but I've observed that the RAM usage increases when I connect from multiple tabs and it increases also when I close those tabs. Seems to be really weird....

**Reply by arpansac** [reply]
Just one more doubt (posted this on gitter too):

How check what's the max limit for a 1CPU 2GB RAM server with the following
Total channels: 13
Total Users: 50

If each user is subscribed to each channel: 50x13= 650 connections
Even if there are approx 50 different streams for each channel to which different or all users are listening (example each 'vote' component is realtime for each message in the chat): 650x50=32500

But I think the information of these streams are maintained in redis (not in memory), am I correct?

How many machines might be needed?
